paddle/fluid/operators/fill_constant_op.h

```diff
@@ -101,29 +102,66 @@ class FillConstantKernel : public framework::OpKernel<T> {
 
     platform::DeviceContextPool &pool = platform::DeviceContextPool::Instance();
     auto &dev_ctx = *pool.Get(ctx.GetPlace());
-    bool cpu_place = force_cpu || ctx.GetPlace() == platform::CPUPlace();
-    if (cpu_place) {
+    int actual_place = place_type;
+
+    if (actual_place == 0) {
+      bool cpu_place = force_cpu || ctx.GetPlace() == platform::CPUPlace();
+      if (cpu_place) {
+        actual_place = 1;
+      }
+#ifdef PADDLE_WITH_CUDA
+      if (!cpu_place) {
+        actual_place = 2;
+      }
+#endif
+#ifdef PADDLE_WITH_XPU
+      if (!cpu_place) {
+        actual_place = 4;
+      }
+#endif
+    }
```

Comment:
Better use ctx.GetPlace() to determine the actual target place, instead of compile flags.

Reply:
you are right, updated.